ca/server.go

```diff
@@ -576,7 +575,22 @@ func (s *Server) signNodeCert(ctx context.Context, node *api.Node) {
 	}
 
 	// Attempt to sign the CSR
-	cert, err := s.securityConfig.RootCA().ParseValidateAndSignCSR(node.Certificate.CSR, node.Certificate.CN, role, s.securityConfig.ClientTLSCreds.Organization())
+	var (
+		cert   []byte
+		rawCSR = node.Certificate.CSR
+		cn     = node.Certificate.CN
+		ou     = role
+		org    = s.securityConfig.ClientTLSCreds.Organization()
+	)
+	switch {
+	case externalCA.CanSign():
+		cert, err = externalCA.Sign(PrepareCSR(rawCSR, cn, ou, org))
+	case rootCA.CanSign():
+		cert, err = rootCA.ParseValidateAndSignCSR(rawCSR, cn, ou, org)
```

Comment:
As suggested above, I think it would be cleaner to always attempt `externalCA.Sign`, and then try `rootCA.ParseValidateAndSignCSR` next if `externalCA.Sign` returned `ErrNoExternalCAURLs`. This means there won't be any time of check/time of use issues.